Sources/Kitura/CodableRouter.swift

```diff
@@ -562,6 +565,25 @@ extension Router {
         }
     }
 
+    internal func isValidParameter(of route: String, isTolerable: Bool) -> Bool {
+        if route.contains(":") {
+            if !isTolerable {
+                Log.error("Erroneous path '\(route)', parameter is not allowed. Codable routes without identifier do not allow parameters.")
+                return false
+            } else {
+                let paramaterString = route.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
+                let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
+                if (parameter.isEmpty || parameter.elementsEqual("id")) {
```

Comment:
This seems like a really long-winded way to express `route.hasSuffix(":id")` ... but I suppose it _does_ also check that the route doesn't also contain arbitrary path parameters.

I think the whole thing could be clearer in its intent expressed (something like) this:
```swift
/// Precondition: Path is known to contain a : character
func pathHasSingleParamIdAsSuffix(_ path: String) -> Bool {
  let paramaterString = route.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
  let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
  return parameter.isEmpty || parameter.elementsEqual("id")
}

func pathSyntaxIsValid(_ path: String, identifierExpected: Bool) -> Bool {
  let parameterSupplied = path.contains(":")
  let validIdentifierSyntaxUsed = parameterSupplied && pathHasSingleParamIdAsSuffix(path)
  if parameterSupplied && !validIdentifierSyntaxUsed {
    // Log error: arbitrary params not allowed
    return false
  }
  if identifierExpected && !parameterSupplied {
    // Warn in this case?
    return true
  }
  if !identifierExpected && identifierSupplied {
    // Log error: id supplied but route doesn't consume one
    return false
  }
  return true
}
```
note, I didn't test this :smile:

Reply:
Thanks @djones6  ..Will address your comments